### PR TITLE
Clean up effort to address test tolerance and complex square root functionality

### PR DESCRIPTION
**Pull request by MitchellThompkins**
## Summary by CodeRabbit

* **New Features**
  * Added complex square-root support and improved eigenvalue accuracy with 2×2 block handling.

* **Refactor**
  * Simplified sqrt implementation and promoted helper utilities to top-level namespace.
  * Unified determinant implementation for consistency.

* **Tests**
  * Standardized and extended test tolerances; added tests for complex-sqrt edge cases and iterative routines.

* **Chores**
  * Updated ignore patterns and development scripts (added Hessenberg Householder helper; removed legacy house implementation).

### CODE REVIEW

**Comment by MitchellThompkins** [comment]
@coderabbitai summary

File: math/functions/sqrt.hpp [hunk 1]
```diff
@@ -53,18 +51,22 @@ template <typename T> constexpr T sqrt_int(const T x)
...
     if (x < static_cast<T>(0))
-        return static_cast<T>(-1);
-    else if (is_float<T>())
-        return internal::sqrt_check(x, static_cast<T>(1));
+    {
+        // We return a poison value (-1) as constexpr NaN is not portably
+        // supported in C++17 without built-ins. In the future, this could
+        // be replaced with a real NaN if a portable constexpr solution is found.
+        return poison_nan<T>();
+    }
```
[the remark this answers is not in the document]
Comment (MitchellThompkins) [reply math/functions/sqrt.hpp]: 04584a2ae87328992cdafd1b0c39274168bd65c5 should have fixed that.

**Comment by MitchellThompkins** [comment]
@coderabbitai summary
